File: analyzer.py
# analyzer.py
# Dewey.Decimal 📚
# A text analysis toolkit for decision-ready insights.

import sys
from collections import Counter
from textblob import TextBlob
import textstat

def analyze_text(file_path):
    with open(file_path, 'r', encoding='utf-8') as f:
        text = f.read()

    # Basic counts
    word_count = len(text.split())
    char_count = len(text)

    # Most common words
    words = [w.lower() for w in text.split() if w.isalpha()]
    common_words = Counter(words).most_common(5)

    # Sentiment
    sentiment = TextBlob(text).sentiment.polarity
    if sentiment > 0.1:
        tone = "Positive"
    elif sentiment < -0.1:
        tone = "Negative"
    else:
        tone = "Neutral"

    # Readability
    readability = textstat.flesch_kincaid_grade(text)

    # Print results
    print(f"📊 Analysis of: {file_path}")
    print(f"Words: {word_count}")
    print(f"Characters: {char_count}")
    print("Most common words:", [w for w, _ in common_words])
    print(f"Sentiment: {tone}")
    print(f"Readability (Grade Level): {readability}")

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Usage: python analyzer.py <file>")
    else:
        analyze_text(sys.argv[1])

File: analyzer.js
// analyzer.js
// Dewey.Decimal 📚
// A text analysis toolkit for decision-ready insights.

const fs = require('fs');
const Sentiment = require('sentiment');
const readability = require('text-readability');

function analyzeText(filePath) {
    const text = fs.readFileSync(filePath, 'utf8');

    // Basic counts
    const words = text.split(/\s+/);
    const wordCount = words.length;
    const charCount = text.length;

    // Most common words
    const freq = {};
    words.forEach(w => {
        w = w.toLowerCase().replace(/[^a-z]/g, '');
        if (w) freq[w] = (freq[w] || 0) + 1;
    });
    const commonWords = Object.entries(freq)
        .sort((a, b) => b[1] - a[1])
        .slice(0, 5)
        .map(([w]) => w);

    // Sentiment
    const sentiment = new Sentiment();
    const sentimentScore = sentiment.analyze(text).score;
    let tone = "Neutral";
    if (sentimentScore > 0) tone = "Positive";
    if (sentimentScore < 0) tone = "Negative";

    // Readability (Flesch-Kincaid)
    const grade = readability.fleschKincaidGradeLevel(text);

    // Print results
    console.log(`📊 Analysis of: ${filePath}`);
    console.log(`Words: ${wordCount}`);
    console.log(`Characters: ${charCount}`);
    console.log("Most common words:", commonWords);
    console.log(`Sentiment: ${tone}`);
    console.log(`Readability (Grade Level): ${grade}`);
}

// CLI
if (process.argv.length < 3) {
    console.log("Usage: node analyzer.js <file>");
} else {
    analyzeText(process.argv[2]);
}
